container-disc/src/main/java/com/yahoo/container/jdisc/ConfiguredApplication.java

```diff
@@ -251,24 +262,31 @@ private void watchPortChange() {
         try {
             while (true) {
                 subscriber.waitNextGeneration(false);
-                QrConfig newConfig = QrConfig.class.cast(first(subscriber.config().values()));
-                reconfigure(qrConfig);
-                if (qrConfig.rpc().port() != newConfig.rpc().port()) {
-                    com.yahoo.protect.Process.logAndDie(
-                            "Rpc port config has changed from " +
-                            qrConfig.rpc().port() + " to " + newConfig.rpc().port() +
-                            ". This we can not handle without a restart so we will just bail out.");
+                if (first(subscriber.config().values()) instanceof QrConfig newConfig) {
+                    reconfigure(newConfig.shutdown());
+                    synchronized (this) {
+                        if (qrConfig.rpc().port() != newConfig.rpc().port()) {
+                            try {
+                                reListenRpc(newConfig);
```

Comment:
Consider adding info logging whenever RPC port changes (if not already present)

Reply:
Done